Integrations/Autodesk/maya/scripts/unityCommands.mel

```diff
@@ -633,13 +668,277 @@ proc setupNewExportSet(string $exportPath, int $exportAnimOnly, string $selected
+            $maxNamespaceCount = $intersectionSize;
+        }
+    }
+    return $commonNamespace;
+}
+
+proc createExportSetDialog(int $exportType){
+    $origSelection = `ls -sl`;
+    if(size($origSelection) <= 0){
+        // nothing selected
+        print ("Nothing selected");
+        return;
+    }
+
+    $exportAnim = false;
+    $exportAnimOnly = false;
+    switch($exportType){
+    case 0 /* export animation only */:
+        $exportAnim = true;
+        $exportAnimOnly = true;
+        break;
+    case 1 /* export model only */:
+        break;
+    default: /* export model + animation */
+        $exportAnim = true;
+        break;
+    }
+
+    // open up a dialog for choosing the export set options
+    string $window = `window -title "Unity FBX Export Options" -iconName "Short Name" -widthHeight 475 250`;
+    
+    string $container = `formLayout -numberOfDivisions 100`;
+    
+    string $mainOptions = `columnLayout -adjustableColumn true -p $container`;
+
+    // go through selection to find common namespace to set as default
+    string $commonNamespace = getCommonNamespace($origSelection);
+    
+    string $modelFilename = "Untitled";
+    // if one item selected, take the name of it as the filename
+    if (size($origSelection) == 1){
+        // take the name of the selection without the namespace
+        string $nTokens[];
+        int $nNumTokens = `tokenize ("" + $origSelection[0]) ":" $nTokens`;
+        $modelFilename = $nTokens[$nNumTokens-1];
+    }
+    else{
+        // if multi items selected, but one of them is the root, then take the name of the root as the filename
+        for($i = 0; $i < size($origSelection); $i++){
+            string $descendents[] = `listRelatives -type "transform" -allDescendents $origSelection[$i]`;
+            string $intersection[] = getIntersection($origSelection, $descendents);
+            if (size($intersection) == size($origSelection)-1){
+                // take the name of the selection without the namespace
+                string $nTokens[];
+                int $nNumTokens = `tokenize ("" + $origSelection[$i]) ":" $nTokens`;
+                $modelFilename = $nTokens[$nNumTokens-1];
+                break;
+            }
+        }
+    }
+
+    int $labelSize = 120;
+    int $textFieldSize = 300;
+
+    // get namespace
+    string $namespaceField = createTextFieldWithLabel("Namespace", $mainOptions, $labelSize, $textFieldSize);
+    textField -e -text $commonNamespace $namespaceField;
+    
+    string $modelFilePath = " 0 0";
+    if(!$exportAnimOnly){
+        string $modelFilePathField = createFilePathField("Model File Path", $mainOptions, $labelSize, $textFieldSize);
+        string $modelFileNameField = createTextFieldWithLabel("Model File Name", $mainOptions, $labelSize, $textFieldSize);
+        textField -e -text ($modelFilename + ".fbx") $modelFileNameField;
+        $modelFilePath = " " + $modelFilePathField + " " + $modelFileNameField;
```

Comment:
have you tested paths with spaces?

Reply:
I haven't good idea. It shouldn't be an issue here though because I am passing the UI's ID not the path itself.

Reply:
@benoithudson just tested and it works with spaces.

Reply:
Oh, I see, you're passing the field itself, not the value of the field. Makes sense!